docs/user/openstack/deploy_sriov_workers.md

```diff
@@ -0,0 +1,380 @@
+  template:
+    metadata:
+      labels:
+        machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+        machine.openshift.io/cluster-api-machine-role: <node_role>
+        machine.openshift.io/cluster-api-machine-type: <node_role>
+        machine.openshift.io/cluster-api-machineset: <infrastructure_ID>-<node_role>
+    spec:
+      providerSpec:
+        value:
+          apiVersion: openstackproviderconfig.openshift.io/v1alpha1
+          cloudName: openstack
+          cloudsSecret:
+            name: openstack-cloud-credentials
+            namespace: openshift-machine-api
+          flavor: <nova_flavor>
+          image: <glance_image_name_or_location>
+          serverGroupID: <optional_UUID_of_server_group>
+          kind: OpenstackProviderSpec
+          networks:
+            subnets:
+            - uuid: <machines_subnet_uuid>
+          ports:
+            - networkID: <radio_network_uuid>
+              nameSuffix: radio
+              fixedIPs:
+                - subnet_id: <radio_subnet_uuid>
+              tags:
+                - sriov
+                - radio
+              vnicType: direct
+              portSecurity: false
+            - networkID: <uplink_network_uuid>
+              nameSuffix: uplink
+              fixedIPs:
+                - subnet_id: <uplink_subnet_uuid>
+              tags:
+                - sriov
+                - uplink
+              vnicType: direct
+              portSecurity: false
+          primarySubnet: <machines_subnet_uuid>
+          securityGroups:
+          - filter: {}
+            name: <infrastructure_ID>-<node_role>
+          serverMetadata:
+            Name: <infrastructure_ID>-<node_role>
+            openshiftClusterID: <infrastructure_ID>
+          tags:
+          - openshiftClusterID=<infrastructure_ID>
+          trunk: false
+          userDataSecret:
+            name: <node_role>-user-data
+          availabilityZone: <optional_openstack_availability_zone>
+```
+
+Once you have finished editing your machineSet, upload it to your OpenShift cluster:
+
+```sh
+c create -f sriov_machineset.yaml
+```
+
+## Creating SR-IOV Worker Nodes in UPI
+
+Since we largely leave UPI implementation up to the customer, there is no official script for deploying SR-IOV worker nodes. However, we will use a verified example based on the [compute-nodes.yaml](../../../upi/openstack/compute-nodes.yaml) script to walk you through the process. To use this script, open up a terminal to the location of `inventory.yaml` and `common.yaml` UPI Ansible scripts. In the example below, we add the provider networks named `radio` and `uplink` that we created before to the `inventory.yaml` file. Note that the count parameter specifies the number of virtual functions (VFs) to attach to each worker node. This code can also be found on [github](https://github.com/shiftstack/SRIOV-Compute-Nodes-Ansible-Automation).
```

Comment:
```suggestion
Because UPI implementation depends largely on your deployment environment and requirements, there is no official script for deploying SR-IOV worker nodes. However, we can share a verified example that is based based on the [compute-nodes.yaml](../../../upi/openstack/compute-nodes.yaml) script to help you understand the process. 

To use the script, open up a terminal to the location of the `inventory.yaml` and `common.yaml` UPI Ansible scripts. In the the following example, we add provider networks named `radio` and `uplink` to the `inventory.yaml` file. 

Note that the count parameter specifies the number of virtual functions (VFs) to attach to each worker node. This code can also be found on [github](https://github.com/shiftstack/SRIOV-Compute-Nodes-Ansible-Automation).
```